DOC: Improve log scale example

timhoffm opened the pull request:
- Remove https://matplotlib.org/stable/gallery/scales/semilogx_demo.html It is redundant to the other log scale example.
- Explain that semilogx/semilogy/loglog are shortcuts.
- Tune data/visualization

scottshambaugh commented:
New example image:
![image](https://github.com/user-attachments/assets/ec5259e9-feac-4583-afb7-393452b59199)

Maybe this is nitpicky, but could we have a "plain" semilogy plot next to the semilogx one? I think the base 2 labeling is super useful to include, but it takes some interpretation. It'd be nice to have a semilogy plot that you can see at a glance what's going on, and then the base 2 labeling plot could be kept but moved down.

timhoffm commented:
Indeed a bit nitpicky. 😄  It's a bit the question of how much can you illustrate with an example.

The [original example](https://matplotlib.org/stable/gallery/scales/log_demo.html) had the base 2 on the loglog plot. I found this very unfortunate because (1) I've never seed double-log plots with different bases. That seems quite exotic. (2) to get that, one had to use `set_xscale` after `loglog`, which is not quite instructive as you override the x log scale defined by loglog. I thought it's nice to keep the *base 2* aspect somewhere, and the semilogy was the one where it was least confusing. But maybe it's better to leave the base 2 out here?

I also think that adding minor grids here would be too cluttered.

If we want to have all the aspects in, we probably need to structure the example into subsections:

1. semilogx, semilogy, loglog
2. base 2
3. minor grids
4. log with errorbars that get negative

scottshambaugh commented:
However you want to slice it! I think it's definitely an improvement over what's there currently.

timhoffm commented:
@scottshambaugh I've reworked the example. There's now one plot with semilogx/semilogy/loglog, also containing minor grids, as that makes it more obvious which directions are logarithmic.

I've added separate examples `base=2` and handling of non-positive values.

scottshambaugh commented:
I think this looks great, breaking it up into sections was a good move and the exposition helps a lot to interpret what's going on. Good to merge unless you want another reviewer!
